easybuild/easyconfigs/j/jax/jax-0.2.28-foss-2021a-CUDA-11.3.1.eb

```diff
@@ -0,0 +1,129 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Denis Kristak
+easyblock = 'PythonBundle'
+
+name = 'jax'
+version = '0.2.28'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://pypi.python.org/pypi/jax'
+description = """Composable transformations of Python+NumPy programs:
+differentiate, vectorize, JIT to GPU/TPU, and more"""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+builddependencies = [
+    ('Bazel', '3.7.2'),
+    ('pytest-xdist', '2.3.0'),
+    # git 2.x required to fetch repository 'io_bazel_rules_docker'
+    ('git', '2.32.0', '-nodocs'),
+]
+
+dependencies = [
+    ('CUDA', '11.3.1', '', True),
+    ('cuDNN', '8.2.1.32', versionsuffix, True),
+    ('NCCL', '2.10.3', versionsuffix),
+    ('Python', '3.9.5'),
+    ('SciPy-bundle', '2021.05'),
+    ('flatbuffers-python', '2.0'),
+    ('typing-extensions', '3.10.0.0'),
+]
+
+# running the tests with lots of cores results in test failures because not enough threads can be started,
+# and running with a single core also results in systematic test failures
+# 4 cores seems to work best
+parallel = 4
+
+# downloading TensorFlow tarball to avoid that Bazel downloads it during the build
+# note: this *must* be the exact same commit as used in jaxlib-*/WORKSPACE
+local_tf_commit = 'e662039f98d2b6a0d8242887d0881bae04e2d590'
+local_tf_dir = 'tensorflow-%s' % local_tf_commit
+local_tf_builddir = '%(builddir)s/' + local_tf_dir
+
+# replace remote TensorFlow repository with the local one from EB
+local_jax_prebuildopts = "sed -i -f jaxlib_local-tensorflow-repo.sed WORKSPACE && "
+local_jax_prebuildopts += "sed -i 's|EB_TF_REPOPATH|%s|' WORKSPACE && " % local_tf_builddir
+
+use_pip = True
+
+default_easyblock = 'PythonPackage'
+default_component_specs = {
+    'sources': [SOURCE_TAR_GZ],
+    'source_urls': [PYPI_SOURCE],
+    'start_dir': '%(name)s-%(version)s',
+    'use_pip': True,
+    'sanity_pip_check': True,
+    'download_dep_fail': True,
+}
+
+components = [
+    ('absl-py', '0.13.0', {
+        'options': {'modulename': 'absl'},
+        'checksums': ['6953272383486044699fd0e9f00aad167a27e08ce19aae66c6c4b10e7e767793'],
+    }),
+    ('jaxlib', '0.1.76', {
+        'sources': [
+            '%(name)s-v%(version)s.tar.gz',
+            {
+                'download_filename': '%s.tar.gz' % local_tf_commit,
+                'filename': 'tensorflow-%s.tar.gz' % local_tf_commit,
+            }
+        ],
+        'source_urls': [
+            'https://github.com/google/jax/archive/',
+            'https://github.com/tensorflow/tensorflow/archive/'
+        ],
+        'patches': [
+            ('jaxlib_local-tensorflow-repo.sed', '.'),
+            'jaxlib-0.1.70_add-bazel-args-to-shutdown.patch',
+            ('TensorFlow-2.7.0_cuda-noncanonical-include-paths.patch', '../' + local_tf_dir),
+        ],
```

Comment:
These must reside in j/jaxlib, not in j/jax. Please fix or wait for the framework change that will (if accepted) allow us to specify a different software tree to look in for the files.

Reply:
I see! thanks, I have the software already installed, so I guess I will wait until the fix is officially in EB.